apitype/validator.go

```diff
@@ -0,0 +1,22 @@
+package apitype
+
+import (
+	"github.com/go-playground/validator/v10"
+)
+
+// NewValidator creates a new validator with ExplicitNullable validation configured.
+//
+// This function is exported so that it can be used by other packages that need
+// to validate Patch fields.
+func NewValidator() *validator.Validate {
+	// WithRequiredStructEnabled can be removed once validator/v11 is released.
+	val := validator.New(validator.WithRequiredStructEnabled())
+	return WithExplicitNullableValidation[string](val)
+}
+
+// WithExplicitNullableValidation registers the validator with the
+// ExplicitNullable type.
+func WithExplicitNullableValidation[T any](val *validator.Validate) *validator.Validate {
```

Comment:
@brandur I realized `NewValidator` above was specifically registering the validator with only the `string` form of `ExplicitNullable` which isn't flexible enough. I extracted this helper to at least make it easier to register additional ones. Not set on the API but good enough to move forward on for now, lmk if you have better ideas!